rust_src/src/frames.rs

```diff
@@ -727,6 +727,43 @@ pub fn make_frame_invisible(frame: LispFrameLiveOrSelected, force: bool) {
     }
 }
 
+#[lisp_fn]
+pub fn last_nonminibuffer_frame() -> LispObject {
+    unsafe {
+        last_non_minibuffer_frame.into()
+    }
+}
+
+/// A frame to a guaranteed existing frame which is not just a mini-buffer,
+/// or None if there are no such frames.
+/// This is usually the most recent such frame that was selected.
+static mut last_non_minibuffer_frame: Option<LispFrameRef> = None;
+
+/// Return current last non-minibuffer frame reference
+/// or a pointer to null in case there are no such frames
+#[no_mangle]
+pub extern "C" fn get_last_nonminibuffer_frame() -> LispFrameRef {
+    unsafe {
+        match last_non_minibuffer_frame {
+            Some(frame) => frame,
+            None => LispFrameRef::new(std::ptr::null_mut()),
+        }
+    }
+}
+
+/// Set current last non-minibuffer frame reference
+#[no_mangle]
+pub extern "C" fn set_last_nonminibuffer_frame(frame: LispFrameRef) {
+    unsafe {
+        // frame reference may be null, since it is called from C
+        if frame.is_null() {
+            last_non_minibuffer_frame = None
+        } else {
+            last_non_minibuffer_frame = Some(frame)
+        }
+    }
+}
```

Comment:
This is may feel a little weird, because optional ref is sort of `Option^2`, because `Option` may be `Some`/`None` and also reference may point to null (`None`) or a real frame (`Some`)

Should we add a doc comment above LispFrameRef declaration stating that LispFrameRef is guaranteed to point to a real frame. Guaranteed by developers, not by the type system. This is due to C interop and this will be fixed later?

Reply:
Why use an `Option` to define `last_non_minibuffer_frame`? If LispFrameRef is a pointer we can simply check it for NULL right?

Reply:
I think we should move away from Frame pointers and we indeed can do that, but for this, we should migrate all code that accesses frame to Rust, afaiu.